dpdata/lammps/dump.py

```diff
@@ -76,16 +88,17 @@ def safe_get_posi(lines,cell,orig=np.zeros(3), unwrap=False) :
     assert coord_tp_and_sf is not None, 'Dump file does not contain atomic coordinates!'
     coordtype, sf, uw = coord_tp_and_sf
     id_idx = keys.index('id') - 2
+    tidx = keys.index('type') - 2
     xidx = keys.index(coordtype[0])-2
     yidx = keys.index(coordtype[1])-2
     zidx = keys.index(coordtype[2])-2
     sel = (xidx, yidx, zidx)
     posis = []
     for ii in blk :
         words = ii.split()
-        posis.append([float(words[id_idx]), float(words[xidx]), float(words[yidx]), float(words[zidx])])
-    posis.sort()
-    posis = np.array(posis)[:,1:4]
+        posis.append([float(words[id_idx]), float(words[tidx]), float(words[xidx]), float(words[yidx]), float(words[zidx])])
```

Comment:
```suggestion
        posis.append([float(words[tidx]), float(words[id_idx]), float(words[xidx]), float(words[yidx]), float(words[zidx])])
```
Then we do not need to implement _sort_tid_comparator

Reply:
`_sort_tid_comparator` is used so that there is only one possibility for sorting.
In this case, we may get multiple orders when sorting. For example:
```
# order1
id  type_id        x      y     z
1       1            1.0   1.0   1.0
2       1            2.0   2.0   2.0
3       2            3.0   2.0   2.0

# order2
id  type_id        x      y     z
2       1            2.0   2.0   2.0
1       1            1.0   1.0   1.0
3       2            3.0   2.0   2.0
```
Would this be OK?

Reply:
Please check:
```
In [18]: a = np.array([[1, 2], [1, 1]])                                                                                                                                               

In [19]: a.sort(axis=0)                                                                                                                                                               

In [20]: a                                                                                                                                                                            
Out[20]: 
array([[1, 1],
       [1, 2]])
```

Reply:
please check
```
In [33]: a = [[1, 2, 1], [1, 1, 3]]                                                                                                                                                   

In [34]: a.sort()                                                                                                                                                                     

In [35]: print(a)                                                                                                                                                                     
[[1, 1, 3], [1, 2, 1]]

```